docs/contributing/contributing_setup.md

```diff
@@ -47,6 +47,49 @@ In order to contribute to Great Expectations, you will need the following:
 
 ### Install Python dependencies
 
+#### (Easy version of steps 5-7 below for Mac/Linux users)
+
+Create a virtual environment in your locally cloned repo, use the same version of `pip` that we use in our CI/CD pipelines (for Python 3.6 - 3.9), and install the fewest dependencies needed for a dev environment (to minimize potential setup headaches).
+
+```
+python3 -m venv ge_dev
+
+source ge_dev/bin/activate
+
+pip install --upgrade pip==21.3.1
+
+pip install -r requirements-dev-lite.txt -c constraints-dev.txt -e .
+```
+
+Confirm that tests are passing (only against pandas and sqlalchemy with sqlite), without the need for running any Docker containers.
+
+```
+ulimit -n 4096
+
+pytest -v --no-spark --no-postgresql
+```
+
+> In your `~/.zshrc` or `~/.bashrc` file, you will want to add `ulimit -n 4096` so that it is already set for future runs. **You WILL eventually see many tests failing with `OSError: [Errno 24] Too many open files`** if you do not set it!
+
+If you want to slowly add other SQL dialects (instead of adding everything at once via `requirements-dev.txt`), you can `pip install` the following
+
+- postgresql: `psycopg2-binary>=2.7.6`
+- mysql: `PyMySQL>=0.9.3,<0.10`
+- mssql: `pyodbc>=4.0.30` (see step 6 for links on getting the odbc driver on your system first)
+- athena: `pyathena>=1.11`
+- bigquery: `sqlalchemy-bigquery>=1.3.0 google-cloud-secret-manager>=1.0.0 google-cloud-storage>=1.28.0`
+- snowflake: `snowflake-connector-python==2.5.0 snowflake-sqlalchemy>=1.2.3 azure-storage-blob>=12.5.0`
+- redshift: `sqlalchemy-redshift>=0.7.7`
+- teradata: `teradatasqlalchemy==17.0.0.1`
+- oracle: `cx_Oracle`
+- dremio: `sqlalchemy-dremio>=1.2.1 pyarrow>=0.12.0 pyodbc>=4.0.30`
```

Comment:
one small suggestion for this @kenwade4 would it be possible to have a reference here to the appropriate `requirements.txt` file?  I imagine this section becoming really quickly out of date otherwise.  I'm imagining something similar to what we do with python scripts in our how-to-guides. 

```python file=../../../../tests/integration/docusaurus/connecting_to_your_data/database/mssql_yaml_example.py#L3-L6
```

Reply:
The requirements listed reference different requirements files and combine them in some instances (i.e. stuff from requirements-dev-base.txt and requirements-dev-sqlalchemy.txt for the bigquery line). The oracle requirement isn't in any of the files since we aren't testing on oracle DBs in CI/CD.

I'd be more inclined to not include that list of individual dialects at all and tell them to "try setting up the full dev environment (as mentioned in step 6) when you're ready for more robust testing of your custom Expectations" or something like that.

I'm pretty sure @donaldheppner is opposed to splitting up the requirements-dev-*.txt files into things like `requirements-dev-postgresql.txt` etc for dialects (although `requirements-dev-spark.txt` only has a single line). If we were to split them up, THEN it would make sense to link to specific lines of files with `python file=../../../blah`

Reply:
Ok @kenwade4 that makes sense. I guess I keep coming back to the thought of the list becoming out of date. As another idea would it be possible to add a (preferably loud) note to the `requirements` files as a reminder to update the `contributing_setup` doc if anything changes?